react-native-template-reason/src/App.re

```diff
@@ -4,55 +4,50 @@
  *
  * Converted from https://github.com/facebook/react-native/blob/d752446b23e0a12171ca0c01d20f5be625f40455/template/App.js
  */
-open BsReactNative;
+open ReactNative;
 
-let instructions =
-  switch (Platform.os()) {
-  | IOS(_) => "Press Cmd+R to reload,\n" ++ "Cmd+D or shake for dev menu"
-  | Android =>
-    "Double tap R on your keyboard to reload,\n"
-    ++ "Shake or press menu button for dev menu"
-  | exception (Platform.UnknownPlatform(p)) when p == "web" => "Press Cmd+R to reload."
-  | exception (Platform.UnknownPlatform(_p)) => ""
-  };
+module Styles = {
+  open Style;
 
-let styles =
-  Style.(
-    StyleSheet.create({
-      "container":
-        style([
-          flex(1.),
-          justifyContent(Center),
-          alignItems(Center),
-          backgroundColor(String("#F5FCFF")),
-        ]),
-      "welcome":
-        style([
-          fontSize(Float(20.)),
-          textAlign(Center),
-          margin(Pt(10.)),
-        ]),
-      "instructions":
-        style([
-          textAlign(Center),
-          color(String("#333333")),
-          marginBottom(Pt(5.)),
-        ]),
-    })
+  let container = style(
+    ~flex=1.,
+    ~justifyContent=`center,
+    ~alignItems=`center,
+    ~backgroundColor="#F5FCFF"
   );
 
-let app = () =>
-  <View
-    style={
-      styles##container;
-    }>
-    <Text style={styles##welcome}>
-      "Welcome to (Bs) React Native!"->ReasonReact.string
+  let welcome = style(
+    ~fontSize=20.,
+    ~textAlign=`center,
+    ~margin=pt(10.)
+  );
+
+  let instructions = style(
+    ~textAlign=`center,
+    ~color="#333333",
+    ~marginBottom=pt(5.)
+  );
+};
+
+[@react.component]
+let app = () => {
+  let instructions =
+    switch(Platform.os) {
+    | ios => "Press Cmd+R to reload,\n" ++ "Cmd+D or shake for dev menu"
+    | android =>
+      "Double tap R on your keyboard to reload,\n"
+      ++ "Shake or press menu button for dev menu"
+    };
```

Comment:
Can we keep example for other platform please (at least web)?

Reply:
Yes I can add that. BTW why the choice of removing variants? I would have expect lowercase polymorphic variants as in the other new modules. Instead we have directly a string type as in JS which means we can write a pattern matching with random string and it would compile. Isn't it a regression compared to what `bs-react-native` did?

Reply:
First take a look at this https://github.com/reasonml-community/bs-react-native/issues/239

Then this is simply to get a zero-cost binding. Using variant create some unecessary js. Polymorphic variant are not an option considering the fact that we cannot use bs.string in this kind of scenario (if I am correct).
So we used bs.inline (requested for this kind of purpose, see https://github.com/BuckleScript/bucklescript/issues/3472)

Reply:
OK, I see.